app/backend/approaches/chatreadretrieveread.py

```diff
@@ -58,14 +92,15 @@ def __init__(self, search_client: SearchClient, chatgpt_deployment: str, chatgpt
         self.chatgpt_token_limit = get_token_limit(chatgpt_model)
 
     def run(self, history: Sequence[dict[str, str]], overrides: dict[str, Any]) -> Any:
-        has_text = overrides.get("retrieval_mode") in ["text", "hybrid", None]
-        has_vector = overrides.get("retrieval_mode") in ["vectors", "hybrid", None]
-        use_semantic_captions = True if overrides.get("semantic_captions") and has_text else False
-        top = overrides.get("top") or 3
-        exclude_category = overrides.get("exclude_category") or None
-        filter = "category ne '{}'".format(exclude_category.replace("'", "''")) if exclude_category else None
+        with tracer.start_as_current_span("ChatReadRetrieveReadApproach.run"):
```

Comment:
Is there an approach that doesnt require adding "with tracer" to each of these functions? Ideally, we could just attach middleware to the routes.